corgi/tasks/rhel_compose.py

```diff
@@ -49,5 +51,12 @@ def get_builds(compose_names: list[str] = None, stream_name: str = None) -> None
     elif stream_name:
         relations_query = relations_query.filter(product_ref=stream_name)
 
-    relations_build_ids = relations_query.values_list("build_id", flat=True).distinct()
-    fetch_modular_builds(relations_build_ids)
+    processed_builds = 0
+    for build_id in relations_query.values_list("build_id", flat=True).distinct().iterator():
+        if not build_id:
+            continue
+        if not SoftwareBuild.objects.filter(build_id=int(build_id)).exists():
+            logger.info("Processing Compose relation build with id: %s", build_id)
+            slow_fetch_modular_build.delay(build_id, force_process=force_process)
```

Comment:
Should we move this logic into the fetch_modular_builds function?

Reply:
I don't think that's a good idea. We want to avoid create a task (and adding it to Redis) if the build has already been processed.

Reply:
I was referring to the helper function ```corgi.tasks.brew.fetch_modular_builds```, not the Celery task ```corgi.tasks.brew.slow_fetch_modular_build```. The logic you added above to ```corgi.tasks.rhel_compose.get_builds``` is the same as in ```corgi.tasks.brew.fetch_unprocessed_relations```.

I thought there might be other places in the code that needed to avoid scheduling Celery tasks for builds that already exist. I thought we should combine that logic in one place (fetch_modular_builds), then reuse it for any code that processes data in the PCR table. But after looking further, there aren't any other places that need this logic, so what you have here is fine.